test/functional/modules/blocks/chain.js

```diff
@@ -0,0 +1,758 @@
+/* eslint-disable mocha/no-pending-tests, mocha/no-skipped-tests */
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+var crypto = require('crypto');
+var lisk = require('lisk-js');
+var accountFixtures = require('../../../fixtures/accounts');
+var randomUtil = require('../../../common/utils/random');
+var application = require('../../../common/application'); // eslint-disable-line no-unused-vars
+
+var previousBlock;
+
+function createBlock(
+	blocksModule,
+	blockLogic,
+	secret,
+	timestamp,
+	transactions
+) {
+	var keypair = blockLogic.scope.ed.makeKeypair(
+		crypto
+			.createHash('sha256')
+			.update(secret, 'utf8')
+			.digest()
+	);
+	blocksModule.lastBlock.set(previousBlock);
+	var newBlock = blockLogic.create({
+		keypair,
+		timestamp,
+		previousBlock: blocksModule.lastBlock.get(),
+		transactions,
+	});
+	newBlock.id = blockLogic.getId(newBlock);
+	newBlock.height = previousBlock ? previousBlock.height + 1 : 1;
+	return newBlock;
+}
+
+describe('blocks/chain', () => {
```

Comment:
- Why test file is not located under `test/functional/system/blocks`?
- Why a test skeleton which describe unit tests? Seems like we only describe block application anyway.